File: Presentations/PEPM2020/projectile2/cpp/Control.cpp
#include <fstream>
#include <iostream>
#include <string>

#include "Calculations.hpp"
#include "InputConstraints.hpp"
#include "InputFormat.hpp"
#include "OutputFormat.hpp"

using std::ofstream;
using std::string;

int main(int argc, const char *argv[]) {
    ofstream outfile;
    string filename = argv[1];
    outfile.open("log.txt", std::fstream::app);
    outfile << "var 'filename' assigned to ";
    outfile << filename;
    outfile << " in module Control" << std::endl;
    outfile.close();
    double v_launch;
    double theta;
    double p_target;
    double g_vect = 9.8;
    double epsilon = 2.0e-2;
    outfile.open("log.txt", std::fstream::app);
    outfile << "var 'g_vect' assigned to ";
    outfile << g_vect;
    outfile << " in module Control" << std::endl;
    outfile.close();
    outfile.open("log.txt", std::fstream::app);
    outfile << "var 'epsilon' assigned to ";
    outfile << epsilon;
    outfile << " in module Control" << std::endl;
    outfile.close();
    get_input(filename, v_launch, theta, p_target);
    input_constraints(v_launch, theta, p_target);
    double t_flight = func_t_flight(v_launch, theta, g_vect);
    outfile.open("log.txt", std::fstream::app);
    outfile << "var 't_flight' assigned to ";
    outfile << t_flight;
    outfile << " in module Control" << std::endl;
    outfile.close();
    double p_land = func_p_land(v_launch, theta, g_vect);
    outfile.open("log.txt", std::fstream::app);
    outfile << "var 'p_land' assigned to ";
    outfile << p_land;
    outfile << " in module Control" << std::endl;
    outfile.close();
    double d_offset = func_d_offset(p_target, p_land);
    outfile.open("log.txt", std::fstream::app);
    outfile << "var 'd_offset' assigned to ";
    outfile << d_offset;
    outfile << " in module Control" << std::endl;
    outfile.close();
    string s = func_s(p_target, epsilon, d_offset);
    outfile.open("log.txt", std::fstream::app);
    outfile << "var 's' assigned to ";
    outfile << s;
    outfile << " in module Control" << std::endl;
    outfile.close();
    write_output(s, d_offset);
    
    return 0;
}

File: Presentations/PEPM2020/projectile2/python/Control.py
import sys

import Calculations
import InputConstraints
import InputFormat
import OutputFormat

filename = sys.argv[1]
outfile = open("log.txt", "a")
print("var 'filename' assigned to ", end='', file=outfile)
print(filename, end='', file=outfile)
print(" in module Control", file=outfile)
outfile.close()
g_vect = 9.8
epsilon = 2.0e-2
outfile = open("log.txt", "a")
print("var 'g_vect' assigned to ", end='', file=outfile)
print(g_vect, end='', file=outfile)
print(" in module Control", file=outfile)
outfile.close()
outfile = open("log.txt", "a")
print("var 'epsilon' assigned to ", end='', file=outfile)
print(epsilon, end='', file=outfile)
print(" in module Control", file=outfile)
outfile.close()
v_launch, theta, p_target = InputFormat.get_input(filename)
InputConstraints.input_constraints(v_launch, theta, p_target)
t_flight = Calculations.func_t_flight(v_launch, theta, g_vect)
outfile = open("log.txt", "a")
print("var 't_flight' assigned to ", end='', file=outfile)
print(t_flight, end='', file=outfile)
print(" in module Control", file=outfile)
outfile.close()
p_land = Calculations.func_p_land(v_launch, theta, g_vect)
outfile = open("log.txt", "a")
print("var 'p_land' assigned to ", end='', file=outfile)
print(p_land, end='', file=outfile)
print(" in module Control", file=outfile)
outfile.close()
d_offset = Calculations.func_d_offset(p_target, p_land)
outfile = open("log.txt", "a")
print("var 'd_offset' assigned to ", end='', file=outfile)
print(d_offset, end='', file=outfile)
print(" in module Control", file=outfile)
outfile.close()
s = Calculations.func_s(p_target, epsilon, d_offset)
outfile = open("log.txt", "a")
print("var 's' assigned to ", end='', file=outfile)
print(s, end='', file=outfile)
print(" in module Control", file=outfile)
outfile.close()
OutputFormat.write_output(s, d_offset)
